test/NuGet.Core.Tests/NuGet.Packaging.Test/BestPracticeRuleTests.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Packaging.Rules;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    //Package authoring best practices are an ever evolving set of rules and guidelines.
+    //Those soft recommendations are non-breaking validation message. 
+    public class BestPracticeRuleTests
```

Comment:
For unit tests, I feel like following the same name is super important because of discoverability.

Reply:
From https://github.com/NuGet/NuGet.Client/pull/5066#discussion_r1120997940

> I created a subfolder BestPractice under Rules folder.

As I commented above in this conversation, best practises evolve over time as we learn from experience. Do we planning on modifying client every time best practises change? Are we planning on creating pack validations for every best practise that can be programantically checked?

In this specific case, having a readme seems like a good idea for packages on nuget.org, and also on private feeds, so I don't see a problem in "forcing" this rule on everyone (but I feel strongly that it should be possible to NoWarn to hide the message. It would actually be much better if we implemented roslyn analyzer style editorconfig settings, so that customers can change defaults about error, warning, info, none on a code by code basis)

Reply:
Thanks! Changed the test class name to `MissingReadmeRuleTests`.

Reply:
According to our previous discussion, this is going to be a quick first step experiment. 
There are concerns that it takes some time for users to adopt the client tooling to get the latest best practices.
We will evaluate if readme adoption increases significantly after this  implementation, then we can decide what would be the next steps.